streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java

```diff
@@ -232,11 +247,19 @@
                     );
                     break;
                 case ROCKS_DB:
-                    supplier = Stores.persistentTimestampedWindowStore(
-                        materialized.storeName(),
-                        Duration.ofMillis(retentionPeriod),
-                        Duration.ofMillis(windows.size()),
-                        false
+                    supplier = emitStrategy.type() == StrategyType.ON_WINDOW_CLOSE ?
+                        RocksDbIndexedTimeOrderedWindowBytesStoreSupplier.create(
+                            materialized.storeName(),
```

Comment:
@guozhangwang , should we call this another name to avoid overriding existing emit eager's store?

Reply:
If a user just enables `emit final` for their existing topology which uses `emit eager`, will it cause it to use existing state store which has wrong data format etc?

Reply:
I think it's ok to declare this a non-supported pattern? We should call it out in the docs.